File: dogpack/apps/1d/quad_moments/shock_tube/plotq1.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%
%%   Information you have available to you:
%%
%%     Basic information:
%%              mx:  number of points
%%    [xlow,xhigh]:  min/max values of grid
%%            meqn:  number of equations
%%            maux:  number of aux components
%%           meth1:  spatial order of accuracy
%%
%%   Grid information:
%%              xc: grid points (cell centers), size = (mx,my)
%%
%%   Solution information:
%%           qsoln:  solution sampled on mesh, size = (mx,meqn)
%%             aux:  aux components sampled on mesh, size = (mx,maux)
%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% desnity
rho = qsoln(:,1);

% velocity
u = qsoln(:,2)./qsoln(:,1);

% pressure
press = qsoln(:,3)-(qsoln(:,2).^2)./qsoln(:,1);

% heat flux
qheat = qsoln(:,4) - 3*press.*u - rho.*u.^3;

figure(1);
clf;
pz=plot(xc,qsoln(:,1),'bo');
set(pz,'linewidth',1);
set(pz,'markersize',6);
hold off;
axis on; box on; grid off;
%axis([0 1 0.8 3.2]);
set(gca,'plotboxaspectratio',[1.5 1 1]);
set(gca,'xtick',-2:0.25:2);
%set(gca,'ytick',-3:0.5:3);
set(gca,'fontsize',16);
t1 = title(['Density at t = ',num2str(time),'     [DoGPack]']); 
set(t1,'fontsize',16);

figure(2);
clf;
pz=plot(xc,qsoln(:,2)./qsoln(:,1),'bo');
set(pz,'markersize',6);
set(pz,'linewidth',1);
hold off;
axis on; box on; grid off;
%axis([0 1 -0.05 0.55]);
set(gca,'plotboxaspectratio',[1.5 1 1]);
set(gca,'xtick',-2:0.25:2);
%set(gca,'ytick',-1:0.25:1);
set(gca,'fontsize',16);
t1 = title(['Velocity at t = ',num2str(time),'     [DoGPack]']); 
set(t1,'fontsize',16);

figure(3);
clf;
pz=plot(xc,press,'bo');
set(pz,'linewidth',1);
set(pz,'markersize',6);
hold off;
axis on; box on; grid off;
%axis([0 1 0.8 3.2]);
set(gca,'plotboxaspectratio',[1.5 1 1]);
set(gca,'xtick',-2:0.25:2);
%set(gca,'ytick',-3:0.5:3);
set(gca,'fontsize',16);
t1 = title(['Pressure at t = ',num2str(time),'     [DoGPack]']); 
set(t1,'fontsize',16);

figure(4);
clf;
pz=plot(xc,qheat,'bo');
set(pz,'linewidth',1);
set(pz,'markersize',6);
hold off;
axis on; box on; grid off;
%axis([0 1 -2 0.5]);
set(gca,'plotboxaspectratio',[1.5 1 1]);
set(gca,'xtick',-2:0.25:2);
%set(gca,'ytick',-3:0.5:3);
set(gca,'fontsize',16);
t1 = title(['Heat flux at t = ',num2str(time),'     [DoGPack]']); 
set(t1,'fontsize',16);

File: dogpack/apps/1d/quad_moments/Riemann_example/plotq1.py

#----------------------------------------------------------
def plotq1(m,meth1,meqn,mx,time,xc,qsoln,auxsoln):
    
    import matplotlib.pyplot as plt
    import numpy as np
    from math import fabs


    rho = qsoln[:,0]

    u = qsoln[:,1]/qsoln[:,0]

    press = qsoln[:,2]-pow(qsoln[:,1],2)/qsoln[:,0]

    qheat = qsoln[:,3] - 3*press*u - rho*pow(u,3)

    r = qsoln[:,4] - 4.0*qheat*u - 6.0*press*pow(u,2) - rho*pow(u,4)     

    qlow  = min(rho)
    qhigh = max(rho)
    if fabs(qhigh-qlow)>1.0e-7:            
        qeps  = 0.1*(qhigh-qlow)
    else:
        qeps = 0.1 
      

    plt.figure(1)
    plt.clf()
    plt.plot(xc,rho,'b-')
    plt.gca().set_xlim([xc[0],xc[mx-1]])
    plt.gca().set_ylim([qlow-qeps,qhigh+qeps])
    tmp1 = "".join(("Density at t = "))
    tmp2 = "".join((tmp1,str(time)))
    title = "".join((tmp2,"     [DoGPack]"))
    plt.title(title)
    plt.draw()

    qlow  = min(u)
    qhigh = max(u)
    if fabs(qhigh-qlow)>1.0e-7:            
        qeps  = 0.1*(qhigh-qlow)
    else:
        qeps = 0.1 

    plt.figure(2)
    plt.clf()
    plt.plot(xc,u,'b-')
    plt.gca().set_xlim([xc[0],xc[mx-1]])
    plt.gca().set_ylim([qlow-qeps,qhigh+qeps])
    tmp1 = "".join(("Velocity at t = "))
    tmp2 = "".join((tmp1,str(time)))
    title = "".join((tmp2,"     [DoGPack]"))
    plt.title(title)
    plt.draw()

    qlow  = min(press)
    qhigh = max(press)
    if fabs(qhigh-qlow)>1.0e-7:            
        qeps  = 0.1*(qhigh-qlow)
    else:
        qeps = 0.1 

    plt.figure(3)
    plt.clf()
    plt.plot(xc,press,'b-')
    plt.gca().set_xlim([xc[0],xc[mx-1]])
    plt.gca().set_ylim([qlow-qeps,qhigh+qeps])
    tmp1 = "".join(("Pressure at t = "))
    tmp2 = "".join((tmp1,str(time)))
    title = "".join((tmp2,"     [DoGPack]"))
    plt.title(title)
    plt.draw()

    qlow  = min(qheat)
    qhigh = max(qheat)
    if fabs(qhigh-qlow)>1.0e-7:            
        qeps  = 0.1*(qhigh-qlow)
    else:
        qeps = 0.1 

    plt.figure(4)
    plt.clf()
    plt.plot(xc,qheat,'b-')
    plt.gca().set_xlim([xc[0],xc[mx-1]])
    plt.gca().set_ylim([qlow-qeps,qhigh+qeps])
    tmp1 = "".join(("Heat at t = "))
    tmp2 = "".join((tmp1,str(time)))
    title = "".join((tmp2,"     [DoGPack]"))
    plt.title(title)
    plt.draw()

    qlow  = min(r)
    qhigh = max(r)
    if fabs(qhigh-qlow)>1.0e-7:            
        qeps  = 0.1*(qhigh-qlow)
    else:
        qeps = 0.1 
   
    plt.figure(5)
    plt.clf()
    plt.plot(xc,r,'b-')
    plt.gca().set_xlim([xc[0],xc[mx-1]])
    plt.gca().set_ylim([qlow-qeps,qhigh+qeps])
    tmp1 = "".join(("r at t = "))
    tmp2 = "".join((tmp1,str(time)))
    title = "".join((tmp2,"     [DoGPack]"))
    plt.title(title)
    plt.draw()
#----------------------------------------------------------
